File: backend/app/schemas/data_quality.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DataQualityIssueCreate(BaseModel):
    issue_type: str
    severity: str = "medium"
    status: str = "open"
    entity_type: str | None = None
    entity_id: uuid.UUID | None = None
    field_name: str | None = None
    description: str
    suggestion: str | None = None
    detected_by: str | None = None


class DataQualityIssueUpdate(BaseModel):
    issue_type: str | None = None
    severity: str | None = None
    status: str | None = None
    entity_type: str | None = None
    entity_id: uuid.UUID | None = None
    field_name: str | None = None
    description: str | None = None
    suggestion: str | None = None
    resolved_by: uuid.UUID | None = None
    resolved_at: datetime | None = None
    resolution_notes: str | None = None
    detected_by: str | None = None


class DataQualityIssueRead(BaseModel):
    id: uuid.UUID
    building_id: uuid.UUID
    issue_type: str
    severity: str
    status: str
    entity_type: str | None
    entity_id: uuid.UUID | None
    field_name: str | None
    description: str
    suggestion: str | None
    resolved_by: uuid.UUID | None
    resolved_at: datetime | None
    resolution_notes: str | None
    detected_by: str | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/dataQuality.ts
import { apiClient } from '@/api/client';
import type { PaginatedResponse } from '@/types';

export interface DataQualityIssue {
  id: string;
  building_id: string;
  issue_type: string;
  severity: string;
  status: string;
  entity_type: string | null;
  entity_id: string | null;
  field_name: string | null;
  description: string;
  suggestion: string | null;
  resolved_by: string | null;
  resolved_at: string | null;
  resolution_notes: string | null;
  detected_by: string | null;
  created_at: string;
}

export interface DataQualityFilters {
  issue_type?: string;
  severity?: string;
  status?: string;
  page?: number;
  size?: number;
}

export const dataQualityApi = {
  list: async (buildingId: string, filters?: DataQualityFilters): Promise<PaginatedResponse<DataQualityIssue>> => {
    const params: Record<string, string | number> = { size: filters?.size ?? 50 };
    if (filters?.issue_type) params.issue_type = filters.issue_type;
    if (filters?.severity) params.severity = filters.severity;
    if (filters?.status) params.status = filters.status;
    if (filters?.page) params.page = filters.page;
    const response = await apiClient.get<PaginatedResponse<DataQualityIssue>>(
      `/buildings/${buildingId}/data-quality-issues`,
      { params },
    );
    return response.data;
  },

  update: async (
    buildingId: string,
    issueId: string,
    data: { status?: string; resolution_notes?: string },
  ): Promise<DataQualityIssue> => {
    const response = await apiClient.put<DataQualityIssue>(
      `/buildings/${buildingId}/data-quality-issues/${issueId}`,
      data,
    );
    return response.data;
  },
};
